etc/testing/travis.sh

```diff
@@ -19,9 +25,54 @@ until kubectl version >/dev/null 2>/dev/null; do
 	sleep 1;
 done
 
+echo "Running test suite based on BUCKET=$BUCKET"
+
+PPS_SUITE=`echo $BUCKET | grep PPS > /dev/null; echo $?`
+
+make docker-build
+make clean-launch-dev
+make launch-dev
+
 go install ./src/server/cmd/match
-echo "Running local tests"
-make test
+
+if [[ "$BUCKET" == "MISC" ]]; then
+	echo "Running misc test suite"
+	make test-misc
+elif [[ $PPS_SUITE -eq 0 ]]; then
+	PART=`echo $BUCKET | grep -Po '\d+'`
+	NUM_BUCKETS=`cat etc/build/PPS_BUILD_BUCKET_COUNT`
+	echo "Running pps test suite, part $PART of $NUM_BUCKETS"
+	LIST=`go test -v  ./src/server/ -list ".*" | grep -v ok | grep -v Benchmark`
+	COUNT=`echo $LIST | tr " " "\n" | wc -l`
+	BUCKET_SIZE=$(( $COUNT / $NUM_BUCKETS ))
+	MIN=$(( $BUCKET_SIZE * $(( $PART - 1 )) ))
+	#The last bucket may have a few extra tests, to accommodate rounding errors from bucketing:
+	MAX=$COUNT 
+	if [[ $PART -ne $NUM_BUCKETS ]]; then
+		MAX=$(( $MIN + $BUCKET_SIZE ))
+    fi
+
+	RUN=""
+	INDEX=0
+
+	for test in $LIST; do
+		if [[ $INDEX -ge $MIN ]]; then
+			if [[ $INDEX -lt $MAX ]]; then
```

Comment:
I think you can write this as `if [[ $INDEX -ge $MIN ]] && [[ $INDEX -lt $MAX]]; then ...`

Reply:
Good point, done